### PR TITLE
Move header items inside card on all dashboard screens

### LINKED ISSUE #4587: Move Buttons
### To do
- [x] move buttons on every sidebar level screen into the card (Clusters, Templates, Backups, SSH Keys, Members, Service Accounts) Example given for: Clusters, Members
- [x] move cluster level buttons into Cluster card, _move Create Machine Deployment button into MD card (already done in another issue)_
- [x] made in-cards buttons colored (primary or secondary style is now "allowed" in cards instead of just outlined teritary style, this means changes on the admin panel, aka making the buttons the primary style -> blue background)
- [x] Move share and delete cluster into the cluster dropdown

Note:
there's two options for MD screen 1) as is now 2) Buttons with labels (I have decided against writing 'Machine Deployment' or 'MD' in every button as it needs too much space and seems unnecessary)

I'm not sure about which one I prefer yet

### Design
https://www.figma.com/file/dJQKgUPzIDND36o2Nq8dct/Loodse-UI?node-id=6756%3A60595
Clusters

Clusters_Overview

### PR DESCRIPTION
**Pull request by Talha-Jamil-TJ**
**What this PR does / why we need it**:

On all dashboard screens, moved buttons on top of cards inside of the cards

#### Resources > Clusters Page
image

#### Resources > External Clusters Page
image

#### Cluster Templates Page
image

#### etcd Backups > Automatic Backups
image

#### etcd Backups > Snapshots
image

#### etcd Backups > Restores
image

#### Access > SSH Keys
image

#### Access > Members
image

#### Access > Groups
image

#### Access > Service Accounts
image

#### Cluster Details Page
image

#### External Cluster Details Page
image

**Which issue(s) this PR fixes**:

Fixes #4587

**What type of PR is this?**
/kind design

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
NONE
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

**Comment by Waseem826** [comment]
@Talha-Jamil-TJ please take a look why integration tests are failing and fix accordingly.

File: modules/web/src/app/cluster/list/external-cluster/template.html [hunk 1]
```diff
@@ -14,33 +14,34 @@
 limitations under the License.
 -->
```
Comment (KhizerRehan) [inline modules/web/src/app/cluster/list/external-cluster/template.html:16]: This comment is for **External Cluster Details** page:


I think layout what you have followed in **Cluster** details page must be followed for **External** details page

**see**
- Move top buttons for details page just like did in cluster details page wdyt? `...` 
- Add Machine Deployment
- Quota Details widgets

missing-in-external-clusters

File: modules/web/src/app/dynamic/enterprise/quotas/quota-widget/template.html [hunk 2]
```diff
@@ -33,7 +33,10 @@
 
   <ng-template #quotaWidgetWithoutCard>
     <div class="km-quota-widget"
-         [class.km-no-quota-top-margin]="!quotaDetails"
+         [ngClass]="{
+          'km-no-quota-top-margin': !quotaDetails,
+          'widget-border': !showAsCard && !showQuotaWidgetDetails && showBorderOutline
```
Comment (KhizerRehan) [inline modules/web/src/app/dynamic/enterprise/quotas/quota-widget/template.html:38]: For readability
```suggestion
          'widget-border': (!showAsCard && !showQuotaWidgetDetails) && showBorderOutline
```

File: modules/web/src/assets/css/global/_theme.scss [hunk 3]
```diff
@@ -216,6 +216,17 @@
     }
   }
 
+  .km-provider-edit-settings {
+    background-color: map.get($colors, option-background);
+    color: map.get($colors, text);
```
Comment (KhizerRehan) [inline modules/web/src/assets/css/global/_theme.scss:221]: i don't think `color` property is requried as color is applied based on ` .mat-menu-item` can you check whether it is required or not?

**Review by KhizerRehan — CHANGES_REQUESTED** [review]
Feedback PTAL

File: modules/web/src/app/cluster/details/cluster/template.html [hunk 4]
```diff
@@ -121,12 +100,27 @@
...
                   [disabled]="!isRevokeTokenEnabled()">
             <span>Revoke Token</span>
           </button>
+          <button mat-menu-item
+                  *ngIf="config.share_kubeconfig"
+                  (click)="shareConfigDialog()">
+            <span>Share Cluster</span>
+          </button>
+          <mat-divider></mat-divider>
+          <span [matTooltip]="!!cluster?.deletionTimestamp ? 'Cluster is being deleted.' : ''">
```
[the remark this answers is not in the document]
Comment (Talha-Jamil-TJ) [reply modules/web/src/app/cluster/details/cluster/template.html:114]: I need to show a tooltip when the button is disabled, but the button is not showing any tooltip in the disabled state, so this is my workaround.

File: modules/web/src/app/dynamic/enterprise/group/template.html [hunk 5]
```diff
@@ -130,23 +128,3 @@
     </div>
   </mat-card-content>
 </mat-card>
-<!-- </km-dynamic-tab> -->
```
[the remark this answers is not in the document]
Comment (Waseem826) [reply modules/web/src/app/dynamic/enterprise/group/template.html:133]: Should be removed in different PR (perhaps as part of https://github.com/kubermatic/dashboard/issues/5345?)

File: modules/web/src/app/cluster/list/external-cluster/template.html (hunk 1 above)
Comment (Talha-Jamil-TJ) [reply modules/web/src/app/cluster/list/external-cluster/template.html:16]: UI is updated 
image

File: modules/web/src/app/dynamic/enterprise/group/template.html (hunk 5 above)
[the remark this answers is not in the document]
Comment (KhizerRehan) [reply modules/web/src/app/dynamic/enterprise/group/template.html]: Ticket is updated

**Pull request merged**